release-22.2: builtins: implement to_char for timestamps and intervals

otan opened the pull request:
Backport 4/4 commits from #91382.

/cc @cockroachdb/release

---

Resolves #91156

These commits implement to_char for timestamp and interval types. Most
of the code is copied from the PG implementation.

Release justification: critical feature addition with minimal blast radius

otan committed:
tree: move duration parsing logic to duration package

To allow parsing intervals in more places, move the duration parsing
logic to the duration package.

Release note: None

otan committed:
tochar: implement roman numerals for to_char

For ~shits and giggles~ completeness!

This commit implements the roman numeral commands for to_char.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/91541)
<!-- Reviewable:end -->

otan committed:
builtins: implement `to_char` for timestamp types

Release note (sql change): The `to_char(timestamp, string)` function
has now been implemented.

otan committed:
builtins: implement to_char for intervals

Release note (sql change): The `to_char(interval, string)` builtin has
now been implemented.
